require/requirements_test.go

```diff
@@ -681,3 +683,31 @@ func TestErrorAssertionFunc(t *testing.T) {
 		})
 	}
 }
+
+func TestEventuallyWithTFalse(t *testing.T) {
+	mockT := new(MockT)
+
+	condition := func(collect *assert.CollectT) {
+		True(collect, false)
+	}
+
+	EventuallyWithT(mockT, condition, 100*time.Millisecond, 20*time.Millisecond)
+	if !mockT.Failed {
+		t.Error("Check should fail")
+	}
+}
+
+func TestEventuallyWithTTrue(t *testing.T) {
+	mockT := new(MockT)
+
+	state := 0
+	condition := func(collect *assert.CollectT) {
+		state += 1
+		True(collect, state == 2)
+	}
+
+	EventuallyWithT(mockT, condition, 100*time.Millisecond, 20*time.Millisecond)
+	if mockT.Failed {
+		t.Error("Check should pass")
+	}
+}
```

Comment:
Add a check of `state` to enforce that the condition is not called after the first success.